src/extension/baser.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Uri, window, workspace } from 'vscode';
+import '../shared/extension';
+import { Store } from './store';
+
+export class Baser {
+    constructor(
+        private readonly distinctLocalNames: Map<string, string>,
+        private readonly store: Pick<Store, 'distinctArtifactNames'>) {
+    }
+
+    private basesArtifactToLocal = new Map<string, string>([]) // <artifactPath, localPath>
```

Comment:
Still missing some semicolons :)

Reply:
Looks like I need to add "@typescript-eslint/member-delimiter-style" on top of the rule "semi". Will do that.

---
In reply to: [441617882](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441617882) [](ancestors = 441617882)

Reply:
Fixed.

---
In reply to: [441707908](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441707908) [](ancestors = 441707908,441617882)